src/tax-lot/tax-lot.service.spec.ts

```diff
@@ -21,6 +24,23 @@ describe("TaxLotController", () => {
     taxLotService = moduleRef.get<TaxLotService>(TaxLotService);
   });
 
+  describe("findTaxLotByBbl", () => {
+    it("should return a tax lot when requesting a valid bbl", async () => {
```

Comment:
One thing you might consider testing in "happy path" tests is `expect`ing that the service calls methods on repos that is "responsible" for calling. For instance, you might expect that your instance of `taxLotRepository` called it's `findByBbl` method with the bbl you passed in. Jest has methods such as `toHaveBeenCalledWith()` for this sort of thing.

Reply:
I place assertions like `toHaveBeenCalled...` under the umbrella of "testing implementation details". Kent C Dodds has an [article explaining how assertions against implementation details can frustrate]() the development and refactoring process.

In the case of services, developers should be confident that they can call the service method with a parameter and it will return a data structure with a specific structure. They shouldn't need to know that it calls repository methods to do this.

Testing "results" and not "implementation details" becomes more powerful when dealing with e2e tests. Take for example a tax lot endpoint. We want to refactor the controller/service/repo methods to remove the 'TaxLot' from the method names. Any 'results' based tests will be handle this refactor without issues. We just need to update the repo mock to match the real repo's refactor. However, any tests that assert that a specific controller, service, or repo calls a specific method will fail, even though the code works as its users expect.